### ISSUE TITLE
How to do a Update?

### ISSUE THREAD

**Issue by porschberg** [crramirez/limesurvey#35]
Hi,
we use your docker-image for a long time now with success. Thank you.
However: Did you have an upgrade-guide? What is when I want to upgrade to a new version of Limesurvey? Currently we use Version 2.72.3+171020.
I fear that a simple `docker-compose build` will not work with my current version of Limesurvey and I want not lose my old surveys.

Thomas

**Comment by porschberg** [comment]
I pulled the new docker-image. Now i get a "LimeSurvey installer" dialogue. Does it recognize the old database?

**Comment by crramirez** [comment]
Yes, put the same parameters it will say that there is an existing database

**Comment by porschberg** [comment]
Good point. When I login into the docker-container I see:
```
root@7303394b11c0:/var/lib/mysql# ls
ib_logfile0  ib_logfile1  ibdata1  limesurvey  mysql  performance_schema

```
So my database is called `limesurvey` probably. Yes?
Can I read the USER/PASSWORD somewhere from the environment? 

I wrote down my password and user, but I'm not sure if it's true.

**Comment by crramirez** [comment]
You can test. If your database name is right it will try to connect. It
won't overwrite your database. But your doubts make me think that you
didn't make a backup first and it is very dangerous to do the update
without a backup.

Anyway in the Readme of the project you have the default values.

**Comment by porschberg** [comment]
I did a backup via the UI and I have now a file `LimeSurvey_limesurvey_dump_2019-09-23.sql` .
In this file I see a INSERT into table `lime_users` with name=admin.

After my first installation of limesurvey I wrote down two user/password pairs:
user: root password: <hidden>
user: admin password: <hidden>
However when I login into my container and try to connect via mysql I get 
`ERROR 1045 (28000): Access denied for user 'admin'@'localhost' (using password: YES) `
The same for user `root`.

This should work at command line?!

**Comment by porschberg** [comment]
Ok, could connect now with user root. The installation process recognizes the old databases but gives an error:
`Die Tabelle "{{template_configuration}}" fÃ¼r die ActiveRecord-Klasse "TemplateConfiguration" kann nicht in der Datenbank gefunden werden.`

English:
The {{template_configuration}} table for the TemplateConfiguration ActiveRecord class can not be found in the database

**Comment by crramirez** [comment]
OK, It allows you to continue? you can go back modifying your docker-compose.yml to 
```
image:
  crramirez/limesurvey:2
```

But probably you will need to restore the backup. And export your surveys and begin from scratch then import the surveys

**Comment by porschberg** [comment]
Okay. I'm back now. I updated in my docker-compose to  `crramirez/limesurvey:2`.
I see my old surveys. I had not to restore my backup. "ComfortUpdate" is not available for me.
How should I proceed? 

When I login into the container and do a
`
mysql> SELECT * FROM template_configuration;
ERROR 1146 (42S02): Table 'limesurvey.template_configuration' doesn't exist
`
This problematic table does still not exist.  Is it possible to upgrade to limesurvey3?

**Comment by crramirez** [comment]
Hello,

You should be in 2.73.1 now, the latest 2.x. You can try the upgrade process again. If it doesn't work, the only way is to export your surveys, start from scratch and import them again.

Regards

**Comment by porschberg** [comment]
Seems to be the way. It is not possible to upgrade from 2 to 3. Same error as before.
